internal/database/postgres.go

```diff
@@ -309,66 +311,125 @@ FROM TempDownsampled;
 }
 
 // TODO: consider returning workstationGroup
-func (conn PostgresConn) LatestData() ([]uplink.GpuStatsUpload, error) {
-	// we pull Uuid twice so we can put one into Stat sample and the other into Info
-	rows, err := conn.db.Query(`SELECT g.Machine, g.Uuid, g.Uuid, g.Name,
-			g.Brand, g.DriverVersion, g.MemoryTotal,
-			s.MemoryUtilisation, s.GpuUtilisation, s.MemoryUsed,
-			s.FanSpeed, s.Temp, s.MemoryTemp, s.GraphicsVoltage,
-			s.PowerDraw, s.GraphicsClock, s.MaxGraphicsClock,
-			s.MemoryClock, s.MaxMemoryClock
+func (conn PostgresConn) LatestData() (broadcast.Workstations, error) {
+	// pull all the machines in, then gpus for those machines
+	// has to all be done in a transaction to avoid race conditions
+	tx, err := conn.db.Begin()
```

Comment:
With primarily Read queries, using a transaction is unnecessary and inefficient. There is no need to enforce read-write race condition freedom as we accept eventual consistency over to-the-instant consistency.

Reply:
Eh, you could end up with some weird anomalies of machines/gpu's moving inside the same read. I don't think DB perf is a major concern, so I'm happy to merge this as is.